ec2/snapshots.go

```diff
@@ -69,3 +69,22 @@ func (e *Ec2) GetSnapshot(ctx context.Context, ids ...string) ([]*ec2.Snapshot,
 
 	return out.Snapshots, nil
 }
+
+func (e *Ec2) CreateSnapshot(ctx context.Context, input *ec2.CreateSnapshotInput) (string, error) {
+	if input == nil {
+		return "", apierror.New(apierror.ErrBadRequest, "invalid input", nil)
+	}
+	log.Infof("creating snapshot: %s", aws.StringValue(input.VolumeId))
+	out, err := e.Service.CreateSnapshotWithContext(ctx, input)
+	if err != nil {
+		return "", common.ErrCode("failed to create snapshot", err)
+	}
+
+	log.Debugf("got output creating snapshot: %+v", out)
+
+	if out == nil || len(aws.StringValue(out.SnapshotId)) == 0 {
```

Comment:
can we also test this case?

Reply:
```
func (m mockEC2Client) CreateSnapshotWithContext(ctx aws.Context, input *ec2.CreateSnapshotInput, opts ...request.Option) (*ec2.Snapshot, error) {
	if m.err != nil {
		return nil, m.err
	}
	return &ec2.Snapshot{SnapshotId: aws.String("1234")}, nil
}
``` 

Above is the mock code that I have written. According to this, the mock function will either return an error or an out object. The line that is not covered by the tests, checks for an invalid out object when the error is nil. To cover this line, i think we need to modify the mock functionality by passing extra flags to the `mockEC2Client` struct. Please let me know your thoughts.

Reply:
```
func (m mockEC2Client) CreateSnapshotWithContext(ctx aws.Context, input *ec2.CreateSnapshotInput, opts ...request.Option) (*ec2.Snapshot, error) {
	if m.err != nil {
		return nil, m.err
	}
	return &ec2.Snapshot{SnapshotId: aws.String("1234")}, nil
}
``` 

Above is the mock code that I have written. According to this, the mock function will either return an error or an out object. The line that is not covered by the tests, checks for an invalid out object when the error is nil. To cover this line, i think we need to modify the mock functionality by passing extra flags to the `mockEC2Client` struct. Please let me know your thoughts.